### PR TITLE
Resource Template Pages now loaded 100% statically

### LINKED ISSUE #522: As a developer I would like all siphon nodes to be rendered statically
this is in an effort to reduce the risk of moving from a dynamic loaded front end to a static one using the graphql queries that are already in place. 

This proof of concept will great improve site performance as well as provide good indicators of how to implement this in Devhub V3

### PR DESCRIPTION
**Pull request by patricksimonian**
## Summary
Converted the resource type pages to be loaded 100% static

## Notable Changes 
- addition of a fantastic spec on searching/filtering features in tests
- filtering is now done via url query params

Fixes #522

### CODE REVIEW

File: app-web/src/templates/resourceType.js [hunk 1]
```diff
@@ -19,142 +21,131 @@ import FilterMenu from '../components/Page/FilterMenu';
...
+  pageContext, // received from gatsby create pages api, view gatsby/createPages.js for more info
+  location,
+}) => {
+  const [sideDrawerToggled, setSideDrawerToggled] = useState(false);
+  const [menuToggled, setMenuToggled] = useState(false);
+  const queryParam = queryString.parse(location.search);
+  let query = []; // the search query ie ?q=foo
+  let results = []; // the search results if any
+  let filters = []; // the filter queury ie ?f=bar&f=foo
+  let windowHasQuery = Object.prototype.hasOwnProperty.call(queryParam, 'q');
```
Comment (ShellyXueHan) [inline app-web/src/templates/resourceType.js]: constant

File: app-web/src/templates/resourceType.js [hunk 2]
```diff
@@ -19,142 +21,131 @@ import FilterMenu from '../components/Page/FilterMenu';
...
+    // convert filters into and array of objects with a key props so that it is similar to DEFAULT_FILTER's schema
+    // this will allow an intersection to work
+    const filtersWithKeys = filters.map(f => ({ key: f }));
+    // get active filters by filter keys
+    const activeFilters = intersectionBy(filterGroups, filtersWithKeys, 'key');
+    resources = filterResources(resources, activeFilters);
+  }
+
+  return (
+    <Layout showHamburger toggleMenu={() => setMenuToggled(!menuToggled)} showMenu={menuToggled}>
```
Comment (ShellyXueHan) [inline app-web/src/templates/resourceType.js]: `menuToggled`

File: app-web/__tests__/components/FilterGroup.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    const Checkbox = getByTestId(`${TEST_IDS.checkbox}-${firstFilterKey}`);
+
+    expect(navigate).not.toHaveBeenCalled();
+
+    fireEvent.click(Checkbox);
+
+    expect(navigate).toHaveBeenCalledWith(`${props.location.pathname}`);
+
+    const SecondCheckbox = getByTestId(`${TEST_IDS.checkbox}-${secondFilterKey}`);
```
Comment (ShellyXueHan) [inline app-web/__tests__/components/FilterGroup.test.js]: break into two

File: app-web/__tests__/components/FilterGroup.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  const secondFilterKey = DEFAULT_FILTERS[1].key;
+
+  it('matches snapshot and when the search is empty (?f= null), clicking on a checkbox calls navigate with /foo/?f=[key]', () => {
+    const { container, getByTestId } = render(<FilterGroup {...props} />);
+
+    expect(container.firstChild).toMatchSnapshot();
+
+    const Checkbox = getByTestId(`${TEST_IDS.checkbox}-${firstFilterKey}`);
+
+    expect(navigate).not.toHaveBeenCalled();
```
Comment (ShellyXueHan) [inline app-web/__tests__/components/FilterGroup.test.js]: do you need to check this?

File: app-web/__tests__/pages/resourceType.test.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+    expect(queryByTestId(NO_RESOURCE_TEST_IDS.container)).toBeInTheDocument();
+  });
+
+  test('when searching, if there are results the number of cards should reduce', () => {
+    queryString.parse.mockReturnValue({});
+    useSearch.mockReturnValue([]);
+    const { rerender, queryAllByText } = render(
+      <ThemeProvider theme={theme}>
+        <ResourceType {...props} location={{ search: '?q=foo' }} />
```
Comment (ShellyXueHan) [inline app-web/__tests__/pages/resourceType.test.js]: location to the next one

File: app-web/src/templates/resourceType.js (hunk 1 above)
Comment (patricksimonian) [reply app-web/src/templates/resourceType.js]: fixed

**Pull request merged**